clippy_lints/src/dereference.rs

```diff
@@ -703,6 +713,14 @@ fn try_parse_ref_op<'tcx>(
     }
 }
 
+// Checks if the adjustments contains a deref of `ManuallyDrop<_>`
+fn adjust_derefs_manually_drop<'tcx>(adjustments: &'tcx [Adjustment<'tcx>], mut ty: Ty<'tcx>) -> bool {
+    adjustments.iter().any(|a| {
+        let ty = mem::replace(&mut ty, a.target);
+        matches!(a.kind, Adjust::Deref(Some(ref op)) if op.mutbl == Mutability::Mut) && is_manually_drop(ty)
+    })
+}
+
```

Comment:
I'm not sure about the logic of this function. The original `ty` passed into the function is never read because from the first iteration it is already replaced with an adjustment.

Also, the doc-comment isn't a doc-comment

Reply:
The first iteration doesn't necessarily happen.

Reply:
The result of `mem::replace` on the first iteration is the passed in value.